cargo-near/src/commands/build_command/docker.rs

```diff
@@ -0,0 +1,226 @@
+        util::handle_step("Performing git checks...", || {
+            match context {
+                BuildContext::Deploy => {
+                    let contract_path: camino::Utf8PathBuf = self.contract_path()?;
+                    // TODO: clone to tmp folder and checkout specific revision must be separate steps
+                    eprintln!(
+                        "\n The URL of the remote repository:\n {}\n",
+                        git_checks::remote_repo_url(&contract_path)?
+                    );
+                    Ok(())
+                }
+                BuildContext::Build => Ok(()),
+            }
+        })?;
+        let mut cloned_repo = util::handle_step(
+            "Cloning project repo to a temporary build site, removing uncommitted changes...",
+            || cloned_repo::ClonedRepo::clone(&self),
+        )?;
+
+        let crate_metadata = util::handle_step(
+            "Collecting cargo project metadata from temporary build site...",
+            || {
+                let cargo_toml_path: camino::Utf8PathBuf = {
+                    let mut cloned_path: std::path::PathBuf = cloned_repo.tmp_contract_path.clone();
+                    cloned_path.push("Cargo.toml");
+                    cloned_path.try_into()?
+                };
+                CrateMetadata::collect(CargoManifestPath::try_from(cargo_toml_path)?)
+            },
+        )?;
+
+        let docker_build_meta =
+            util::handle_step("Parsing and validating `Cargo.toml` metadata...", || {
+                metadata::ReproducibleBuild::parse(&crate_metadata)
+            })?;
+
+        util::print_step("Running docker command step...");
+        let (status, docker_cmd) = self.docker_subprocess_step(docker_build_meta, &cloned_repo)?;
+
+        if status.success() {
+            util::print_success("Running docker command step (finished)");
+            // TODO: make this a `ClonedRepo` `copy_artifact` method
+            cloned_repo.tmp_contract_path.push("target");
+            cloned_repo.tmp_contract_path.push("near");
+
+            let dir = cloned_repo.tmp_contract_path.read_dir().wrap_err_with(|| {
+                format!(
+                    "No artifacts directory found: `{:?}`.",
+                    cloned_repo.tmp_contract_path
+                )
+            })?;
+
+            for entry in dir.flatten() {
+                if entry.path().extension().unwrap().to_str().unwrap() == "wasm" {
+                    let wasm_path = {
+                        let mut contract_path = cloned_repo.contract_path.clone();
+                        contract_path.push("contract.wasm");
+                        contract_path
+                    };
+                    std::fs::copy::<std::path::PathBuf, camino::Utf8PathBuf>(
+                        entry.path(),
+                        wasm_path.clone(),
+                    )?;
+
+                    // TODO: ensure fresh
```

Comment:
TODO (please, keep track of all TODOs)

Reply:
transferred to local task-tracking system